### PR TITLE
security: upgrade kafka-clients to 3.6.1

### LINKED ISSUE #6092: 2.0.0 版本引用kafka存在Apache Kafka Clients JNDI注入漏洞 (CVE-2023-25194)，望修复
2.0.0 版本引用kafka-clients-3.1.2.jar存在Apache Kafka Clients JNDI注入漏洞 (CVE-2023-25194)，望修复

### PR DESCRIPTION
**Pull request by slievrly**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

upgrade kafka-clients to 3.5.2

### Ⅱ. Does this pull request fix one issue?

fixes #6092

### CODE REVIEW

File: dependencies/pom.xml [hunk 1]
```diff
@@ -91,6 +91,7 @@
         <logback.version>1.2.9</logback.version>
         <logstash-logback-encoder.version>6.5</logstash-logback-encoder.version>
         <kafka-appender.version>0.2.0-RC2</kafka-appender.version>
+        <kafka-clients.version>3.5.2</kafka-clients.version>
```
Comment (funky-eyes) [inline dependencies/pom.xml]: Why not use 3.6.1?
```suggestion
        <kafka-clients.version>3.6.1</kafka-clients.version>
```

File: changes/zh-cn/2.x.md [hunk 2]
```diff
@@ -27,6 +27,7 @@
 
 ### security:
 - [[#6069](https://github.com/seata/seata/pull/6069)] 升级Guava依赖版本，修复安全漏洞
+- [[#6147](https://github.com/seata/seata/pull/6147)] 升级 kafka-clients依赖至3.5.2
```
Comment (funky-eyes) [inline changes/zh-cn/2.x.md]:
```suggestion
- [[#6147](https://github.com/seata/seata/pull/6147)] 升级 kafka-clients依赖至3.6.1
```

File: changes/en-us/2.x.md [hunk 3]
```diff
@@ -28,6 +28,7 @@ Add changes here for all PR submitted to the 2.x branch.
 
 ### security:
 - [[#6069](https://github.com/seata/seata/pull/6069)] Upgrade Guava dependencies to fix security vulnerabilities
+- [[#6147](https://github.com/seata/seata/pull/6147)] upgrade kafka-clients to 3.5.2
```
Comment (funky-eyes) [inline changes/en-us/2.x.md]:
```suggestion
- [[#6147](https://github.com/seata/seata/pull/6147)] upgrade kafka-clients to 3.6.1
```

File: dependencies/pom.xml (hunk 1 above)
Comment (slievrly) [reply dependencies/pom.xml]: all resolved.

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**